core/src/test/scala/pureconfig/PureconfSuite.scala

```diff
@@ -413,6 +413,40 @@ class PureconfSuite extends FlatSpec with Matchers with OptionValues with TryVal
     loadConfig[ConfWithListOfFoo](conf) shouldBe Success(expected)
   }
 
+  it should "be able to load a set of Ints from the system properties" in {
+    val conf = ConfigFactory.parseString("""
+    pure.conf: {
+      intSet.0: 1
+      intSet.1: 2
+    }""")
+
+    val expected = Set(1, 2)
+    loadConfig[Set[Int]](conf, "pure.conf.intSet") shouldBe Success(expected)
+  }
+
+  it should "be able to load a list of Ints from the system properties in correct order" in {
+    val conf = ConfigFactory.parseString("""
+    pure.conf: {
+      intList.2: 1
+      intList.0: 2
+      intList.1: 3
+    }""")
+
+    val expected = List(2, 3, 1)
+    loadConfig[List[Int]](conf, "pure.conf.intList") shouldBe Success(expected)
+  }
+
+  it should "be able to load a list of Ints from the system properties in correct order when one element is missing" in {
+    val conf = ConfigFactory.parseString("""
+    pure.conf: {
+      intList.2: 3
+      intList.0: 1
+    }""")
+
+    val expected = List(1, 3)
```

Comment:
Gosh, I find this behavior surprising. I'd expect that either a missing index would be an error or result in a `null` (though a may loathe) them, in that index.

That isn't to say that this behavior is wrong; it just wouldn't be my first choice.